fmp_server.py

```diff
@@ -846,13 +851,23 @@ def sendNtcCircleMatchStart(self, seq):
         circle = self.session.get_circle()
         circle.departed = True
 
-        count = circle.get_population()
-        data = struct.pack(">I", count)
+        count = 0
+        data = b''
         for i, player in circle.players:
-            data += struct.pack(">B", i+1)
-            data += pati.lp2_string(player.capcom_id)
-            data += pati.lp2_string(b"\1")
-            data += struct.pack(">H", 21)  # TODO: Field??
+            if player.is_circle_standby():
+                data += struct.pack(">B", i+1)
+                data += pati.lp2_string(player.capcom_id)
+                data += pati.lp2_string(b"\1")
+                data += struct.pack(">H", 21)  # TODO: Field??
+                count += 1
+            else:
+                # Client ignore field
+                ntc_circle_kick = struct.pack(">B", 0) + pati.lp2_string('')
+                pat_handler = self.server.get_pat_handler(player)
+                pat_handler.send_packet(PatID4.NtcCircleKick, ntc_circle_kick,
```

Comment:
I think the bug I mentioned resides here. We should probably remove the kicked player from the circle.

Maybe doing something like this would solve the issue:
```python
kick_list = []
for i, player in circle.players:
    if not player.is_circle_standby():
        kick_list.append(player)
        continue
    data +=  # Rest of the code
    # (...)

ntc_circle_kick = struct.pack(">B", 0) + pati.lp2_string('')
for player in kick_list:
    pat_handler = self.server.get_pat_handler(player)
    pat_handler.send_packet(PatID4.NtcCircleKick, ntc_circle_kick, seq)
    circle.players.remove(player)
```

Reply:
Yep, you are right. I forgot to remove the player from the circle's player collection.

So I modified the `Players::remove` method so it can accept the index, for a more direct removal of the player.